src/rgw/rgw_sync_module_pubsub.cc

```diff
@@ -753,32 +747,25 @@ class PSSubscription {
     }
   };
 
-  using PushCR = RGWPostRESTResourceCR<rgw_pubsub_event, int>;
-
   class StoreEventCR : public RGWCoroutine {
-    RGWDataSyncEnv *sync_env;
-    PSSubscriptionRef sub;
-    EventRef event;
-    PSEvent pse;
-    PSConfigRef& conf;
-    PSSubConfigRef& sub_conf;
-    rgw_object_simple_put_params put_obj;
-    string oid_prefix;
-    int i;
+    RGWDataSyncEnv* const sync_env;
+    const PSSubscriptionRef sub;
+    const PSEvent pse;
+    const string oid_prefix;
+
   public:
-    StoreEventCR(RGWDataSyncEnv *_sync_env,
-                 PSSubscriptionRef& _sub,
-                 EventRef& _event) : RGWCoroutine(_sync_env->cct),
+    StoreEventCR(RGWDataSyncEnv* const _sync_env,
+                 const PSSubscriptionRef& _sub,
+                 const EventRef& _event) : RGWCoroutine(_sync_env->cct),
                                      sync_env(_sync_env),
                                      sub(_sub),
-                                     event(_event),
                                      pse(_event),
-                                     conf(sub->env->conf),
-                                     sub_conf(sub->sub_conf) {
-      oid_prefix = sub->sub_conf->data_oid_prefix;
+                                     oid_prefix(sub->sub_conf->data_oid_prefix) {
     }
 
     int operate() override {
+      // TODO: in case of "push-only" subscription no need to store event
+      rgw_object_simple_put_params put_obj;
```

Comment:
why was this moved from being a class member. Object will be destroyed after first yield. Should be a class member.